File: spring-jpa/jpa_programming/jpa-basic/src/main/java/hellojpa/Member.java
package hellojpa;

import javax.persistence.*;
import java.util.Date;

//@Entity
public class Member {

    public Member() {

    }

    @Id
    private String id;

    @Column(name = "name")
    private String userName;

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getUserName() {
        return userName;
    }

    public void setUserName(String userName) {
        this.userName = userName;
    }
}

File: spring-jpa/kotlin-springboot-jpa-1/src/main/kotlin/com/jpabook/jpashop/domain/Member.kt
package com.jpabook.jpashop.domain

import javax.persistence.*

@Entity
class Member(
    var name: String,

    @Embedded
    var address: Address? = null,

    @OneToMany(mappedBy = "member")
    var orders: MutableList<Order> = mutableListOf(),

    @Id @GeneratedValue
    @Column(name = "member_id")
    var id: Long? = null
)
